subprojects/snapshots/src/main/java/org/gradle/internal/snapshot/impl/DirectorySnapshotter.java

```diff
@@ -200,18 +199,29 @@ public FileVisitResult visitFile(Path file, @Nullable BasicFileAttributes attrs)
                     // when FileVisitOption.FOLLOW_LINKS, we only get here when link couldn't be followed
                     throw new FileSnapshottingException(String.format("Could not list contents of '%s'. Couldn't follow symbolic link.", file));
                 }
-                addFileSnapshot(file, name, attrs);
+                builder.visit(snapshotFile(file, name, attrs));
             }
             return FileVisitResult.CONTINUE;
         }
 
+        private FileSystemLocationSnapshot snapshotFile(Path absoluteFilePath, String name, BasicFileAttributes attrs) {
```

Comment:
just some refactors, to simplify review of important changes in the next commits